packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts

```diff
@@ -54,13 +55,35 @@ export default function useDetectSSOCode(
   // state is used during OAuth based Synapse account creation (it's the username)
   const state = searchParams?.get('state')
 
+  const { accessToken } = useSynapseContext()
```

Comment:
This is the tricky bit. If logged into Synapse and you're logging into ORCiD, then assume this is to bind the ORCiD to your account.  If not logged in and logging into ORCiD, then assume user is attempting to login via ORCiD to an account that is already bound to an ORCiD account

Reply:
Clever!